applications/asset_tracker_v2/src/modules/modem_module.c

```diff
@@ -244,6 +244,28 @@ static void lte_evt_handler(const struct lte_lc_evt *const evt)
 	case LTE_LC_EVT_LTE_MODE_UPDATE:
 		nw_mode_latest = evt->lte_mode;
 		break;
+	case LTE_LC_EVT_MODEM_EVENT:
+		LOG_DBG("Modem domain event, type: %s",
+			evt->modem_evt == LTE_LC_MODEM_EVT_LIGHT_SEARCH_DONE ?
+				"Light search done" :
+			evt->modem_evt == LTE_LC_MODEM_EVT_SEARCH_DONE ?
+				"Search done" :
+			evt->modem_evt == LTE_LC_MODEM_EVT_RESET_LOOP ?
+				"Reset loop" :
+			evt->modem_evt == LTE_LC_MODEM_EVT_BATTERY_LOW ?
+				"Low battery" :
+			evt->modem_evt == LTE_LC_MODEM_EVT_OVERHEATED ?
+				"Modem is overheated" :
+				"Unknown");
+
+		if (evt->modem_evt == LTE_LC_MODEM_EVT_RESET_LOOP) {
+			LOG_WRN("The Modem has detected a reset loop. LTE network attach is now "
+				"restricted for the next 30 minutes. Reboot the device or call the "
+				"XFACTORYRESET AT command to reset the modem to factory settings. "
```

Comment:
**suggestion:** add information explaining what to do in case this happens on a device that is not under manual control of the developer.

Reply:
Please remove the suggestion to use `XFACTORYRESET` as it's not something developers should implement. The fix would be to gracefully shut down the modem before reset. 5 ungraceful resets is what triggers this condition.

@coderbyheart This error typically happens on a developers desk when the developer has an error that triggers a quick reboot loop. After 5 ungraceful resets the modem goes into the `RESET LOOP` state which prevents LTE attach. If this happens on a device in the field I don't think there's much you can do other than hope it doesn't reboot again in the next 30 minutes.

Reply:
I agree. We should probably not recommend user to call the XFACTORYRESET command as it resets other settings in the modem as well.